src/lib.rs

```diff
@@ -31,11 +31,33 @@ extern crate proc_macro;
 mod parse;
 
 use proc_macro::TokenStream;
-use quote::{format_ident, quote};
-use syn::parse_macro_input;
+use proc_macro2::{Ident, Span};
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse_macro_input, Lifetime, LifetimeParam, TypeParamBound};
 
 use crate::parse::Input;
 
+/// Wrapper around syn structs that don't implement `Copy` but we want to use at multiple places
+#[derive(Clone, Copy)]
+struct CopyWrapper<'a, T>(&'a T);
```

Comment:
How does this actually work?  AFAIS the derived `Copy` implementation has a `T: Copy` bound and references to a `ToTokens` implementation also implement `ToTokens`, so why can’t we use a reference directly?

Reply:
Hmm, it looks  like it's either cloning or taking by reference, which means the copywrapper is not required. Removing it just works.

But I remember hitting a case where it didn't.